### PR TITLE
Upgrade from Rails 6.1.6 to 7.0.3

### PR DESCRIPTION
**Pull request by callumknights**
Upgrades from Rails 6.1.6 to 7.0.3.

See individual commits for further details.

[Trello card](https://trello.com/c/cOkOCfta/12-upgrade-transition-from-rails-61x-to-70x)

### CODE REVIEW

**Review by JonathanHallam — COMMENTED** [review]
Looks good, just one small question about gem changes

File: Gemfile.lock [hunk 1]
```diff
@@ -54,8 +54,8 @@ GEM
     activerecord (7.0.2.4)
       activemodel (= 7.0.2.4)
       activesupport (= 7.0.2.4)
-    activerecord-import (1.0.6)
-      activerecord (>= 3.2)
+    activerecord-import (1.4.0)
```
[the remark this answers is not in the document]
Comment (callumknights) [reply Gemfile.lock:57]: Split and explained as discussed in slack 👍

File: features/mapping_history.feature [hunk 2]
```diff
@@ -23,8 +23,8 @@ Feature: History of edits to a mapping
     And I visit the path /sites/directgov/mappings
     And I click the link "Edit"
     And I click the tab "History"
-    Then I should see that New URL was changed from http://somewhere.gov.uk to http://bad.gov.uk
-    And I should see "New URL updated"
+    Then I should see that New Url was changed from http://somewhere.gov.uk to http://bad.gov.uk
```
Comment (JonathanHallam) [inline features/mapping_history.feature:26]: Minor, these don't need changing, it's [captured by a regex](https://github.com/alphagov/transition/blob/be8f60aaa80efbed920daf629dc6f34b8e19b3a2/features/step_definitions/mappings_assertion_steps.rb#L39-L45) in the step definitions

Comment (callumknights) [reply features/mapping_history.feature:26]: This was actually needed to correct a failing test, I was very surprised

**Review by brucebolt — COMMENTED** [review]
Looks good 👍 .

I've just made a few small comments.

File: Gemfile [hunk 3]
```diff
@@ -1,6 +1,6 @@
 source "https://rubygems.org"
 
-gem "rails", "6.1.6"
+gem "rails", "7.0.2.4"
```
[the remark this answers is not in the document]
Comment (callumknights) [reply Gemfile:3]: This keeps happening 😆 Will update 👍

**Review by JonathanHallam — DISMISSED** [review]
nice one, cheers

File: app/views/bulk_add_batches/new.html.erb [hunk 4]
```diff
@@ -11,7 +11,7 @@
 <%= form_tag site_bulk_add_batches_path(@site), role: 'form', 'data-module' => 'toggle-mapping-form-fields' do %>
 
   <% if @batch.errors %>
-    <%= render 'shared/error_messages', error_messages: @batch.errors.values.flatten(1) %>
+    <%= render 'shared/error_messages', error_messages: @batch.errors.map { |error| error.message }.flatten(1) %>
```
Comment (JonathanHallam) [inline app/views/bulk_add_batches/new.html.erb:14]: not sure what the preferred syntax is here, but could use `map(&:message)` here to cut off some characters, like I said, don't actually mind which is used, i just like the &: syntax :D

Comment (brucebolt) [reply app/views/bulk_add_batches/new.html.erb]: Good idea, I've updated the relevant commit.

**Pull request merged**